programs/mango-v4/src/instructions/perp_edit_market.rs

```diff
@@ -74,112 +70,138 @@ pub fn perp_edit_market(
             oracle_price.to_num(),
             Clock::get()?.unix_timestamp.try_into().unwrap(),
         );
+        require_group_admin = true;
     }
 
-    // unchanged -
-    // bids
-    // asks
-    // event_queue
-    // quote_lot_size
-    // base_lot_size
-
     if let Some(maint_base_asset_weight) = maint_base_asset_weight_opt {
         perp_market.maint_base_asset_weight = I80F48::from_num(maint_base_asset_weight);
+        require_group_admin = true;
     }
     if let Some(init_base_asset_weight) = init_base_asset_weight_opt {
+        let old_init_base_asset_weight = perp_market.init_base_asset_weight;
         perp_market.init_base_asset_weight = I80F48::from_num(init_base_asset_weight);
+
+        // security admin can only reduce init_base_asset_weight
```

Comment:
for untrusted operation we also need to check >= 0, sorry only thought of that just now

Reply:
wdym by that? so something like

![image](https://user-images.githubusercontent.com/89031858/214033485-562334a4-68eb-4c7b-81b7-c2b4e3fcce16.png)

Reply:
yeah, this or just generally requiring >= 0. Negative weights don't make sense. We can trust the admin, but not the security authority.

Reply:
I think I will apply it generally